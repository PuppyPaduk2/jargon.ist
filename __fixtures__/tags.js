// @flow

export default [{ title: 'tag1', slug: 'tag1' }, { title: 'tag2', slug: 'tag2' }];
