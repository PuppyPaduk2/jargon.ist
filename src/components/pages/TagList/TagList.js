// @flow

import React, { Fragment, PureComponent } from 'react';
import Link from 'gatsby-link';

import { Container, Section } from '../../ui';

import groupByFirstLetter, { type GroupedItem } from '../../../utils/groupByFirstLetter';
import createChunks from '../../../utils/chunks';

import { type Tag } from '../../../types/tag';

type Props = {
  tags: Array<Tag>,
};

type State = {
  chunks: Array<Array<GroupedItem<Tag>>>,
};

class TagList extends PureComponent<Props, State> {
  constructor(props: Props) {
    super(props);

    const groupedJargons = groupByFirstLetter(props.tags, letter => letter.toLocaleUpperCase());

    this.state = {
      chunks: createChunks(groupedJargons, 4),
    };
  }

  render() {
    const { chunks } = this.state;

    return (
      <Fragment>
        <Section className="u-bg-steel-gray">
          <Container>
            <h1 className="u-color-primary u-clear-gap-top u-gap-bottom-small">Konular</h1>
            <p className="u-color-comet u-half-width@md-up u-full-width@md-down">
              Belli konular için gruplanmış jargonları görüntüleyin.
            </p>
          </Container>
        </Section>

        <Section className="u-pad-bottom-large">
          <Container>
            <div className="row">
              {chunks.map((chunk, index) => (
                /* eslint-disable-next-line react/no-array-index-key */
                <div key={index} className={`col col--lg-${12 / chunks.length}`}>
                  {chunk.map(group => (
                    <div key={group.letter} className="u-pad-bottom-large">
                      <h2 className="u-color-primary u-clear-gap-top">{group.letter}</h2>
                      {group.items.map(tag => (
                        <div key={tag.slug} className="u-gap-bottom-2xsmall">
                          <Link to={`/k/${tag.slug}`} className="c-link-secondary">
                            {tag.title}
                          </Link>
                        </div>
                      ))}
                    </div>
                  ))}
                </div>
              ))}
            </div>
          </Container>
        </Section>
      </Fragment>
    );
  }
}

export default TagList;
