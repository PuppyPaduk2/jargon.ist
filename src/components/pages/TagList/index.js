// @flow

import TagList from './TagList';

export default TagList;
