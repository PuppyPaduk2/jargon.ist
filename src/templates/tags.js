// @flow

import React, { Fragment } from 'react';
import Helmet from 'react-helmet';

import TagList from '../components/pages/TagList';

import { type Tag } from '../types/tag';

type Props = {
  pathContext: {
    tagList: Array<Tag>,
  },
};

export const Tags = ({ pathContext }: Props) => (
  <Fragment>
    <Helmet>
      <title>Konular</title>
    </Helmet>

    <TagList tags={pathContext.tagList} />
  </Fragment>
);

export default Tags;
