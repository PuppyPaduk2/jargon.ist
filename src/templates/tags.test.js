// @flow

import React from 'react';
import { shallow } from 'enzyme';

import { Tags } from './tags';

import TagList from '../components/pages/TagList';

it('renders correctly', () => {
  const wrapper = shallow(<Tags pathContext={{ tagList: [] }} />);

  expect(wrapper).toMatchSnapshot();
});

it('renders title', () => {
  const wrapper = shallow(<Tags pathContext={{ tagList: [] }} />);

  expect(wrapper.find('title')).toHaveLength(1);
});

it('renders TagList component', () => {
  const wrapper = shallow(<Tags pathContext={{ tagList: [] }} />);

  expect(wrapper.find(TagList)).toHaveLength(1);
});
