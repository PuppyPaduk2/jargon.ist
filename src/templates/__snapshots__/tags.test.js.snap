// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<React.Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Konular
    </title>
  </HelmetWrapper>
  <TagList
    tags={Array []}
  />
</React.Fragment>
`;
